### DISCUSSION TITLE
Changing the appearance of the results list

### DISCUSSION THREAD

**Discussion by CleitonDeLima** [codingjoe/django-select2#150] [Ideas]
Hi,

Today django-select2 has some limitations for using the [templateResult](https://select2.org/dropdown#templating) option of select2.
The api is returning by default for each object `id` and `text`. I think it would be interesting to have the option to add more information, in case it is necessary to change the template via the frontend with other information.
There could be something like this:  

```python
class AuthorWidget(s2forms.ModelSelect2Widget):
    data_attrs = ["name", "info"]
    search_fields = [
        "username__icontains",
        "email__icontains",
    ]
```
and the API return could be like this:  

```json
[
  {
    "id": 1,
    "text": "Author str(obj)",
    "data": {
      "name": "Author name",
      "info": "foo" 
    }
  },
  ...
]
```

Example:  
![image](https://user-images.githubusercontent.com/4956372/181352752-ec070765-4ede-43c8-9746-2c5f223a9695.png)

if it's something that makes sense to have I can open a PR.

**Comment by codingjoe** [comment]
Hi @CleitonDeLima,

Interesting idea. I have been using custom Select2 implementations in the past, that were displaying much more information. Does something like that, work out-of-the-box now, or would it require a lot of JS work on our (this pages) side?

Cheers,
Joe

**Reply by CleitonDeLima** [reply]
Hi @codingjoe 

From what I've seen you need a javascript function that returns the template (with jquery) for each customization done.
Unfortunately it is not possible to use django templates. :/
